src/parseWithPointers.ts

```diff
@@ -260,11 +278,65 @@ const transformDuplicatedMappingKeys = (nodes: YAMLNode[], lineMap: number[]): I
   return validations;
 };
 
-const reduceMergeKeys = (items: unknown): object | null => {
+const reduceMergeKeys = (items: unknown, sortKeys: boolean): object | null => {
   if (Array.isArray(items)) {
-    // reduceRight is on purpose here! We need to respect the order - the key cannot be overridden..
-    return items.reduceRight((merged, item) => Object.assign(merged, item), {});
+    // reduceRight is on purpose here! We need to respect the order - the key cannot be overridden
+    const reduced = items.reduceRight(
```

Comment:
Do you have a sense of how this sortKeys option affects performance?

Reply:
I'll check it.
Frankly speaking, though, I wouldn't expect to use it anywhere else than Spectral CLI.

Reply:
Well... I would expect this to be true - `xyz === yaml.stringify(yaml.parse(xyz))`. Is this not the case right now?

Reply:
`xyz === yaml.stringify(yaml.parse(xyz))`

This will be true, but `Object.keys(xyz)[0] === Object.keys(yaml.stringify(yaml.parse(xyz)))[0]` won't if such a YAML document is provided
```yaml
"foo": true
"0": false
```

```ts
Object.keys(xyz)[0] // equals "foo"
Object.keys(yaml.stringify(yaml.parse(xyz)))[0] // equals "0"
```

Reply:
xyz is a string right? So `Object.keys(xyz)` doesn't apply. 

If key ordering is not preserved, I don't understand how `xyz === yaml.stringify(yaml.parse(xyz))` could be true?

Reply:
Ah sorry, I meant to write `Object.keys(yaml.parse(xyz))[0] // equals "foo"` :sweat_smile: 

> If key ordering is not preserved, I don't understand how xyz === yaml.stringify(yaml.parse(xyz)) could be true?

The key ordering should be preservered during stringification, *but* it won't be upon the second parsing unless `preserveOrderKey` is passed again.

I screwed the example there. Here is a correct one.
```
Object.keys(yaml.parse(xyz, { preserveKeyOrder: true }))[0] === Object.keys(yaml.parse(yaml.stringify(yaml.parse(xyz, { preserveKeyOrder: true }))))[0]
```

Reply:
You can also add a test if you'd like to
```ts
test('preserves the order of keys', () => {
  const xyz = `foo: true
bar: false
"1": false
"0": true
`;

  expect(safeStringify(parseWithPointers(xyz, { preserveKeyOrder: true }).data)).toEqual(
    safeStringify(parseWithPointers(safeStringify(parseWithPointers(xyz, { preserveKeyOrder: true }).data), {
      preserveKeyOrder: true,
    }).data as any),
  ); // this assertion is met

  expect(safeStringify(parseWithPointers(xyz, { preserveKeyOrder: true }).data)).toEqual(
      safeStringify(parseWithPointers(safeStringify(parseWithPointers(xyz, { preserveKeyOrder: true }).data)).data as any),
  ); // this assertion is not met
});
```

Reply:
Right but wouldn't one expect `myString === JSON.stringify(JSON.parse(myString))` to be true? 

Take Studio for example - if user opens YAML file, we parse it into memory and then stringify and write it back to disk. What it sounds like is that what is written back to disk will not equal what was originally on disk (key ordering not maintained).

Reply:
> Take Studio for example - if user opens YAML file, we parse it into memory and then stringify and write it back to disk. What it sounds like is that what is written back to disk will not equal what was originally on disk (key ordering not maintained).

It depends on the value of `preserveKeyOrder` and the YAML document itself (if it has no numeric scalar mapping keys, everything will be okay).

Reply:
Alternatively, could add simple `files` option to graphite, similar to tsconfig - https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#details.

> Files included using "include" can be filtered using the "exclude" property. However, files included explicitly using the "files" property are always included regardless of "exclude".